test/unit/retabled/internal_test.cljc

```diff
@@ -0,0 +1,128 @@
+    (testing "Correctly updates when default is set and values are nil"
+      (entries-test 20 {:paging {:entries-option {:default 20}}}))
+    (testing "Correctly updates when default is nil and values are set"
+      (entries-test 3 {:paging {:entries-option {:values [3 20 100]}}}))
+    (testing "Correctly updates when default is set and values are set"
+      (entries-test 20 {:paging {:entries-option {:default 20 :values [3 20 100]}}}))
+    #_(testing "Correctly updates when default is nil and first value is nil" ;; not quite built to handle this yet
+      (entries-test (:entries-per-page @internal/PAGING-INFO) {:paging {:entries-option {:values [nil]}}}))))
+
+(deftest update-current-data!
+  (let [current-data-test (fn [expected table-id entries]
+                            (internal/update-current-data! table-id entries)
+                            (let [actual @internal/CURRENT-DATA]
+                              (is (= expected actual))))]
+    #_(testing "Correctly updates when table-id or entries are nil" ;; Should handle nils
+      (current-data-test @internal/CURRENT-DATA nil nil)
+      (current-data-test @internal/CURRENT-DATA "1" nil)
+      (current-data-test @internal/CURRENT-DATA nil [{:first "entry"}{:second "entry"}]))
+    #_(testing "Correctly updates when table-id and entries are set"
+      (current-data-test {"1" [{:first "entry"}{:second "entry"}]} "1" [{:first "entry"}{:second "entry"}]))))
+
+(deftest update-all-data!
+  (let [all-data-test (fn [expected table-id entries]
+                            (internal/update-all-data! table-id entries)
+                            (let [actual @internal/ALL-DATA]
+                              (is (= expected actual))))]
+    #_(testing "Correctly updates when table-id or entries are nil" ;; Should handle nils
+      (all-data-test @internal/ALL-DATA nil nil)
+      (all-data-test @internal/ALL-DATA "1" nil)
+      (all-data-test @internal/ALL-DATA nil [{:first "entry"}{:second "entry"}]))
+    #_(testing "Correctly updates when table-id and entries are set"
+      (all-data-test {"1" [{:first "entry"}{:second "entry"}]} "1" [{:first "entry"}{:second "entry"}]))))
+
+(deftest generate-table-id
+  (let [table-id-test (fn [expected table-id]
+                        (is (= expected (internal/generate-table-id table-id))))
+        generate-id (fn [] (str "__retabled-" (+ 1 @internal/TABLE-NAME-COUNT)))]
+    (testing "Correctly generates new table-id"
+      (table-id-test (generate-id) nil))
+    (testing "Correctly assigns provided table-id"
+      (let [id (generate-id)]
+        (table-id-test id id)))
+    (testing "Correctly assigns data structure as table-id"
+      (let [id (generate-id)]
+        (table-id-test ["vector" "of" "strings" id] ["vector" "of" "strings" id])))
+    #_(testing "Correctly handles duplicates"))) ;; doesn't detect duplicates because it checks the document for existing table-id's, not just generated table-id's
```

Comment:
Make a test of its own, that includes the situation of handling "what if the web page has tables that are not retabled?"